addons/block_code/ui/blocks/parameter_block/parameter_block.gd

```diff
@@ -16,16 +17,24 @@ var param_name_input_pairs: Array
 var param_input_strings: Dictionary  # Only loaded from serialized
 var spawned_by: ParameterOutput
 
+var _panel_normal: StyleBox
+var _panel_focus: StyleBox
+
 
 func _ready():
 	super()
 
+	_panel_normal = _panel.get_theme_stylebox("panel").duplicate()
+	_panel_normal.bg_color = color
+	_panel_normal.border_color = color.darkened(0.2)
+
+	_panel_focus = _panel.get_theme_stylebox("panel").duplicate()
+	_panel_focus.bg_color = color
+	_panel_focus.border_color = Constants.FOCUS_BORDER_COLOR
```

Comment:
This is indeed awkward since there are now 3 styleboxes including the one saved in the scene. Is it possible to have one stylebox in the code that duplicates the one in the scene and then just mutate the color properties on it instead of overriding back and forth? You could even go further and build the initial override in code instead of taking it from the scene, but I feel like people would prefer to edit the non-color properties in the editor.

Reply:
> This is indeed awkward since there are now 3 styleboxes including the one saved in the scene. Is it possible to have one stylebox in the code that duplicates the one in the scene and then just mutate the color properties on it instead of overriding back and forth? You could even go further and build the initial override in code instead of taking it from the scene, but I feel like people would prefer to edit the non-color properties in the editor.

For reference, other Control nodes in Godot have several styleboxes, one per each style. For example Button has styleboxes for Normal, Pressed, Disabled, Focus.
![Captura desde 2024-08-01 10-49-20](https://github.com/user-attachments/assets/af36ac19-e7c0-4f75-86a9-61cc524b1cd0)

Reply:
Oooh, should we attach a bunch of different styleboxes and then use `theme_type_variation` to switch between them? Right now I have it swapping the stylebox for the "Panel" style, but looking at the documentation for that property I see…

> The name of a theme type variation used by this Control to look up its own theme items. When empty, the class name of the node is used (e.g. Button for the Button control), as well as the class names of all parent classes (in order of inheritance).

… which helps me to understand why styleboxes have these two completely different naming schemes.

Reply:
That's certainly very interesting. I suppose that would be good because we'd be inheriting whatever Panel theme variation is in the project. I'm going to punt and keep what you have for now, but feel free to keep investigating how to better manage this!